sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala

```diff
@@ -47,7 +47,13 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
   def ioschema: ScriptTransformationIOSchema
 
   protected lazy val inputExpressionsWithoutSerde: Seq[Expression] = {
-    input.map(Cast(_, StringType).withTimeZone(conf.sessionLocalTimeZone))
+    input.map { in =>
+      in.dataType match {
+        case _: ArrayType | _: MapType | _: StructType =>
+          new StructsToJson(in).withTimeZone(conf.sessionLocalTimeZone)
```

Comment:
Is it okay to follow the default behaviour w/o `options`?

Reply:
> Is it okay to follow the default behaviour w/o `options`?

I am not so familiar about this part's properties. (I am checking these properties)
Just a quick thought,  how about we add `inputSerdeProps` as `StructTJson`'s properties and add `outputSerdeProps` as `JsonToStruct` 's perperties.

Reply:
> Just a quick thought, how about we add inputSerdeProps as StructTJson's properties and add outputSerdeProps as JsonToStruct 's perperties.

Ah, it seems fine. Could you try it?

Reply:
> > Just a quick thought, how about we add inputSerdeProps as StructTJson's properties and add outputSerdeProps as JsonToStruct 's perperties.
> 
> Ah, it seems fine. Could you try it?

Updated.